apps/astarte_appengine_api/lib/astarte_appengine_api_web/router.ex

```diff
@@ -19,106 +19,171 @@ defmodule Astarte.AppEngine.APIWeb.Router do
   use Astarte.AppEngine.APIWeb, :router
   alias Astarte.AppEngine.APIWeb.Plug.LogRealm
 
-  pipeline :api do
+  pipeline :auth_api do
     plug :accepts, ["json"]
     plug LogRealm
     plug Astarte.AppEngine.APIWeb.Plug.AuthorizePath
   end
 
+  pipeline :device_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogDeviceId
+  end
+
+  pipeline :device_group_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogGroupName
+  end
+
+  pipeline :device_alias_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogDeviceAlias
+  end
+
+  pipeline :interface_value_api do
+    plug Astarte.AppEngine.APIWeb.Plug.JoinPath
+    plug Astarte.AppEngine.APIWeb.Plug.LogInterface
+    plug Astarte.AppEngine.APIWeb.Plug.LogPath
+  end
+
+  pipeline :interface_values_group_api do
+    plug Astarte.AppEngine.APIWeb.Plug.JoinPath
+    plug Astarte.AppEngine.APIWeb.Plug.LogInterface
+    plug Astarte.AppEngine.APIWeb.Plug.LogPath
+  end
+
+  pipeline :interface_values_device_group_api do
+    plug Astarte.AppEngine.APIWeb.Plug.JoinPath
+    plug Astarte.AppEngine.APIWeb.Plug.LogInterface
+    plug Astarte.AppEngine.APIWeb.Plug.LogPath
+  end
+
+  pipeline :device_group_name_api do
+    plug Astarte.AppEngine.APIWeb.Plug.LogDeviceId
+    plug Astarte.AppEngine.APIWeb.Plug.LogGroupName
+  end
```

Comment:
Since the plugs of the `device_group_name_api` pipelines are only applied by the pipelines `device_api` and `device_group_api`, we could use those two instead (structuring the API routes with appropriate scopes).
So then we can remove the definition of `device_group_name_api`.